src/fetchtastic/downloader.py

```
@@ -715,454 +760,280 @@ def check_for_prereleases(
-                            f"Invalid version format in directory {dir_name}, removing"
-                        )
-                        should_keep = False
-                    elif dir_name in repo_firmware_dirs:
-                        # Check if it's newer than the latest release
-                        comparison_result = compare_versions(
-                            dir_version, latest_release_version
-                        )
-                        if comparison_result > 0:
-                            should_keep = True
+    os.makedirs(prerelease_dir, exist_ok=True)
+    real_prerelease_dir = os.path.realpath(prerelease_dir)
```

Ruff (0.13.1)

Local variable `real_prerelease_dir` is assigned to but never used

Remove assignment to unused variable `real_prerelease_dir`

(F841)